app/form_models/publishers/job_listing/applying_for_the_job_form.rb

```diff
@@ -24,4 +29,12 @@ def override_enable_job_applications_for_local_authority!
 
     self.enable_job_applications = false
   end
+
+  # If a Publisher publishes a vacancy for a job role that does not allow enabling job applications
+  # but then changes the job role to one that does, we don't want them to be able to allow applications
+  # but we do want the validations for the enable_job_applications field to pass to prevent an error
+  # from being displayed on the review page.
+  def override_enable_job_applications_when_vacancy_published!
+    self.enable_job_applications = false if vacancy.published? && self.enable_job_applications.nil?
```

Comment:
```ruby
self.enable_job_applications ||= false if vacancy.published?
```

Reply:
I've made a change to this. It seems this works if `self.enable_job_applications` is nil, which it will be when the previous job role didn't allow applications (as the field isn't rendered). When validate_all_steps is run, the value of `enable_job_applications` is successfully overridden.

However, if the user then goes to the applying for the job step and changes something, the validation fails. This is because the value submitted for `enable_job_applications` is now a blank string (as this field is now a hidden field?). 

To fix this, instead of the above, it's now:

```
  def override_enable_job_applications_when_vacancy_listed!
    return if params[:current_organisation].nil?

    self.enable_job_applications = false if enable_job_applications.blank?
  end
```